components/dash-table/src/dash-table/dash/Sanitizer.ts

```diff
@@ -65,6 +66,11 @@ const getFixedRows = (
           (filter_action !== TableAction.None ? 1 : 0) +
           data2number(fixed.data);
 
+const populateColumnsFromData = (data: Data) =>
```

Comment:
Not sure, but maybe we'd like to preserve the order of the columns we are passing to the object generator. It's not really possible using an array of objects, as object properties will always be ordered alphabetically.
So columns will always be displayed in an alphabetic order this way.

One possible solution for this could be instead of passing data from python as `DataFrame.to_dict('records')`, passing it as `DataFrame.to_dict('split')` that contains the headers as an array of strings that preserves column order.

Reply:
I think we're OK here - I believe `Object.keys` now preserves some sort of order. 
```
> Object.keys(JSON.parse('[{"sepal_length": 5.1, "sepal_width": 3.5, "petal_length": 1.4, "petal_width": 0.2, "species": "setosa", "species_id": 1}, {"sepal_length": 4.9, "sepal_width": 3.0, "petal_length": 1.4, "petal_width": 0.2, "species": "setosa", "species_id": 1}, {"sepal_length": 4.7, "sepal_width": 3.2, "petal_length": 1.3, "petal_width": 0.2, "species": "setosa", "species_id": 1}, {"sepal_length": 4.6, "sepal_width": 3.1, "petal_length": 1.5, "petal_width": 0.2, "species": "setosa", "species_id": 1}, {"sepal_length": 5.0, "sepal_width": 3.6, "petal_length": 1.4, "petal_width": 0.2, "species": "setosa", "species_id": 1}]')[0])

Array(6) [ "sepal_length", "sepal_width", "petal_length", "petal_width", "species", "species_id" ]
```

Reply:
We can revisit later, but let's roll with this for now.

Reply:
The JSON serializer we're using in Python sorts keys as of plotly.py 5.0 (whether or not you have `orjson` present), so yes I expect we are going to lose column order this way.